src/controllers/admin/users.js

```diff
@@ -0,0 +1,180 @@
+import usersService from '../../services/users';
+import userValidation from '../../validation/createRole';
+import applicationError from '../../utils/errorHandling/applicationError';
+import userBadRequest from '../../utils/errorHandling/userBadRequest';
+import notFound from '../../utils/errorHandling/notFound';
+import roleServices from '../../services/roles';
+import accessDenied from '../../utils/errorHandling/accessDenied';
+
+exports.findThem = async (req, res, next) =>{
+    try{
+
+
+        const page = req.query.page || 1;
+        const limit = req.query.limit || 3;
+        const skip = ((page - 1) === -1) ? 0 : (page -1) * limit;
+
+
+        //find users using services
+        const users = await usersService.findUsers({
+            offset: skip,
+            limit:limit,
+            attributes:["id","first_name","last_name","username","bio","email","user_role_id","address","language","profile_picture","manager_id"]
+            
+        });
+
+
+        if(users){
+        
+            if(!users.rows.length){ throw new notFound(`No user found on page ${page}`); }
+            const findLineManagers = await roleServices.findLineManagers({});
+            if(findLineManagers){users.allLineManagers = findLineManagers;}
+           return res.status(200).json({status:200, users: users}); 
+        }
+        else{
+            throw new applicationError('Failed to fetch users, try again!', 500);
+        }
+    }
+    catch(error){
+        next(error);
+    }
+}
+
```

Comment:
general feedback, you might need to do a middleware for checking user role and if user is a uthenticated . after if user role and permission is guaranted you should continue to controller , please do have one controller file inclusing all these CRUD operation for permission and role . also do consider make reusable function you will need more than once .